### ISSUE TITLE
stereographic projection question

### ISSUE THREAD

**Issue by aaschwanden** [matplotlib/basemap#97]
Dear Jeff,

I've been using basemap/netCDF4python/griddata for a few years now, and must say that they are some of the best and most useful tools I've ever come across. Thank you very much for your great work!

There is a question, however, that's been puzzling me for a long time now.

My data usually covers Greenland (or parts of it), and is in a stereographic projection, either "+proj=stere +lat_0=90 +lat_ts=70 +lon_0=-45 +k=1 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m +no_defs " (EPSG 3413) or "+proj=stere +lat_0=90 +lat_ts=71 +lon_0=-39 +k=1 +x_0=0 +y_0=0 +ellps=WGS84 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs" (no EPSG code).

Now I would like to set up a basemap instance based on the above projection definition, but where the center of the plot; (lon_0, lat_0) in basemap terms, is given by the center of the actual data I have. And not by (lon_0, lat_0) as given in the projection definition. I don't know how to do this.

What works is: I set up a basemap instance with

```
m = Basemap(width=width,
            height=height,
            resolution='i',
            projection='stere',
            lat_0=lat_0,
            lon_0=lon_0)
```

where width, height, are taken from the data, and (lon_0, lat_0) is the center coordinate of the data. This basemap instance is a stereographic projection which is somewhat different to the data. Therefore I have to convert the data into the basemap projection

 `xx, yy = m(lons, lats)`

where lons and lats are the longitudes and latitudes of my data points. Then I can plot it with

 `m.pcolormesh(xx, yy, data)`

Fine. Except that the data doesn't properly fill the plot area anymore, because it had to be rotated.

So, in conclusion: Is there a way to set up a basemap instance with the exact projection information given above, but where the center of the plot corresponds to the center of the data (and not of the lon_0, lat_0 of the projection)?

![good](https://f.cloud.github.com/assets/120698/84425/eb4cf5a2-640f-11e2-9b38-f53ff78647ee.png)
![bad](https://f.cloud.github.com/assets/120698/84426/eb6afdc2-640f-11e2-8db5-403e2f52ce0f.png)

The script and data to produce the attached figures can be downloaded from http://pism-docs.org/download/stere_plot.zip

Cheers,

Andy

Dr. Andy Aschwanden
Postdoctoral Fellow
Geophysical Institute
903 Koyukuk Drive
Fairbanks AK 99775
USA

email   aaschwanden@alaska.edu
phone +1 (907) 474-7199
fax      +1 (907) 474-7290
